[OpenXR][GTK][WPE] TestWebCore.GraphicsContextGLTextureMapperTest/AnyContextAttributeTest.WebXRBlitTest/* fail

fujii opened the pull request:
#### 994571cc50755b9608be0e26c94921dd3d190558
<pre>
[OpenXR][GTK][WPE] TestWebCore.GraphicsContextGLTextureMapperTest/AnyContextAttributeTest.WebXRBlitTest/* fail
<a href="https://bugs.webkit.org/show_bug.cgi?id=299730">https://bugs.webkit.org/show_bug.cgi?id=299730</a>

Reviewed by Sergio Villar Senin.

An API test for WebXR was failing due to missing required OpenGL
features for WebXR. Enabled the required features. BGRA_EXT doesn&apos;t
seem to be supported. Replaced it with RGBA8.

Test: Tools/TestWebKitAPI/Tests/WebCore/glib/GraphicsContextGLTextureMapper.cpp

* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.cpp:
(WebCore::GraphicsContextGLTextureMapperANGLE::enableRequiredWebXRExtensions):
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.h:
* Tools/TestWebKitAPI/Tests/WebCore/glib/GraphicsContextGLTextureMapper.cpp:
(TestWebKitAPI::TEST_P):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: <a href="https://commits.webkit.org/301194@main">https://commits.webkit.org/301194@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/17c5b2baa8de4b35ff2c7b0f4f97bec524f96f79

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124564 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44237 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34970 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131399 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76519 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126441 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44944 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52811 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94920 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62843 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127518 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35827 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111391 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75331 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34764 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29545 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74881 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105581 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29777 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134067 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51420 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39241 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103231 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51828 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107613 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103013 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48378 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26639 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48356 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19608 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51291 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57084 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50696 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54052 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52380 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

svillar commented on Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.cpp:
```diff
@@ -483,6 +483,18 @@ void GraphicsContextGLTextureMapperANGLE::enableFoveation(GCGLuint)
+        && enableExtension("GL_EXT_discard_framebuffer"_s)
+        && enableExtension("GL_OES_EGL_image"_s)
+        && enableExtension("GL_OES_rgb8_rgba8"_s);
+}
```
I initially added this code here but @carlosgcampos asked me to move it to th GBM GCGL because we can only be sure that DMABuf is available if GBM is available.

In any case to me it still makes sense to have it here and remove it from the GBM file.

fujii replied:
The reason I added extensions here is that I use GraphicsContextGLTextureMapperANGLE for API tests rather than GraphicsContextGLTextureMapperGBM. That's because it doesn't seem to be handy to create  a GraphicsContextGLTextureMapperGBM. I'm not sure which GraphicsContextGLTextureMapperANGLE or GraphicsContextGLTextureMapperGBM should I use for the API tests.

svillar replied:
@carlosgcampos would you agree on moving the implementation to the ANGLE class ? Even though you asked me to move it to the GBM one it might be interesting from the POV of the testing.

carlosgcampos replied:
Yes, I don't remember why I asked you to move it to GBM.

fujii replied:
Thank you for the feedback. Fixed.

svillar replied:
> Yes, I don't remember why I asked you to move it to GBM.

Because you mentioned that the requirements to use DMABuf were only satisfied when GBM was available

carlosgcampos replied:
But there's nothing related to dmabuf in those checks, no?

carlosgcampos replied:
I guess I meant:

```
bool GraphicsContextGLTextureMapperGBM::enableRequiredWebXRExtensions()
{
    if (!makeContextCurrent())
	return false;

    return enableExtension("GL_OES_EGL_image"_s)
        && enableExtension("GL_OES_EGL_image_external"_s)
        && enableExtension("EGL_KHR_image_base"_s)
        && enableExtension("EGL_EXT_image_dma_buf_import"_s)
        && enableExtension("EGL_KHR_surfaceless_context"_s);
}
```

But those are different extensions that are only required for GBM implementation

svillar reviewed (approved):
Makes sense.

fujii committed:
[OpenXR][GTK][WPE] TestWebCore.GraphicsContextGLTextureMapperTest/AnyContextAttributeTest.WebXRBlitTest/* fail
https://bugs.webkit.org/show_bug.cgi?id=299730

Reviewed by Sergio Villar Senin.

An API test for WebXR was failing due to missing required OpenGL
features for WebXR. Enabled the required features. BGRA_EXT doesn't
seem to be supported. Replaced it with RGBA8.

Test: Tools/TestWebKitAPI/Tests/WebCore/glib/GraphicsContextGLTextureMapper.cpp

* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.cpp:
(WebCore::GraphicsContextGLTextureMapperANGLE::enableRequiredWebXRExtensions):
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.h:
* Tools/TestWebKitAPI/Tests/WebCore/glib/GraphicsContextGLTextureMapper.cpp:
(TestWebKitAPI::TEST_P):
* Tools/TestWebKitAPI/glib/TestExpectations.json:

Canonical link: https://commits.webkit.org/301194@main
